client/driver/lxc.go

```diff
@@ -302,21 +313,31 @@ func (d *LxcDriver) Start(ctx *ExecContext, task *structs.Task) (*StartResponse,
 
 	for _, mnt := range mounts {
 		if err := c.SetConfigItem("lxc.mount.entry", mnt); err != nil {
-			return nil, fmt.Errorf("error setting bind mount %q error: %v", mnt, err)
+			return nil, fmt.Errorf("error setting bind mount %q error: %v", mnt, err), c.Destroy
 		}
 	}
 
 	// Start the container
 	if err := c.Start(); err != nil {
-		return nil, fmt.Errorf("unable to start container: %v", err)
+		return nil, fmt.Errorf("unable to start container: %v", err), c.Destroy
+	}
+
+	stopAndDestroyCleanup := func() error {
+		if err := c.Stop(); err != nil {
+			return err
```

Comment:
Should we attempt to Destroy the container even if Stop fails, or will Destroy never succeed if Stop errors? I'm unfamiliar with lxc's behavior in this circumstance and the Go library's docs aren't very useful.

If it's safe to call Destroy if Stop errors, I'd suggest just ignoring Destroy's return value and returning the original Stop effort (so Destroy is just a best-effort at cleaning up).